samples/seldon/mnist_tf.py

```diff
@@ -0,0 +1,222 @@
+             serving_repo='https://github.com/kubeflow/example-seldon.git',
+             serving_branch='master',
+             serving_files='./example-seldon/models/tf_mnist/runtime/*',
+             docker_repo_serving='seldonio/deepmnistclassifier_runtime',
+             docker_tag_serving='0.3'):
+
+#will be pushing image so need docker secret
+#create local with `kubectl create secret generic docker-config --from-file=config.json=${DOCKERHOME}/config.json --type=kubernetes.io/config`
+    secret = k8s_client.V1Volume(
+        name="docker-config-secret",
+        secret=k8s_client.V1SecretVolumeSource(secret_name=docker_secret)
+    )
+
+#use volume for storing model
+    modelvolop = dsl.VolumeOp(
+        name="modelpvc",
+        resource_name="modelpvc",
+        size="50Mi",
+        modes=dsl.VOLUME_MODE_RWO
+    )
+#and another as working directory between steps
+    wkdirop = dsl.VolumeOp(
+        name="wkdirpvc",
+        resource_name="wkdirpvc",
+        size="50Mi",
+        modes=dsl.VOLUME_MODE_RWO
+    )
+
+#clone the training code and move to workspace dir as kaniko (next step) expects that
+    clone = dsl.ContainerOp(
+        name="clone",
+        image="alpine/git:latest",
+        command=["sh", "-c"],
+        arguments=["git clone --depth 1 --branch "+str(training_branch)+" "+str(training_repo)+"; cp "+str(training_files)+" /workspace; ls /workspace/;"],
+        pvolumes={"/workspace": wkdirop.volume}
+    )
+
+#build and push image for training
+    build = dsl.ContainerOp(
+        name="build",
+        image="gcr.io/kaniko-project/executor:latest",
+        arguments=["--dockerfile","Dockerfile","--destination",str(docker_repo_training)+":"+str(docker_tag_training)],
+        pvolumes={"/workspace": clone.pvolume,"/root/.docker/": secret}
+    )
+
+    tfjobjson = """
+{
+	"apiVersion": "kubeflow.org/v1beta1",
+	"kind": "TFJob",
+	"metadata": {
+		"name": "mnist-train-{{workflow.uid}}"
+	},
+	"spec": {
+		"tfReplicaSpecs": {
+			"Worker": {
+				"replicas": 1,
+				"template": {
+					"spec": {
+						"containers": [
+							{
+								"image": "{{workflow.parameters.docker-repo-training}}:{{workflow.parameters.docker-tag-training}}",
+								"name": "tensorflow",
+								"volumeMounts": [
+									{
+										"mountPath": "/data",
+										"name": "persistent-storage"
+									}
+								]
+							}
+						],
+						"restartPolicy": "OnFailure",
+						"volumes": [
+							{
+								"name": "persistent-storage",
+								"persistentVolumeClaim": {
+									"claimName": "{{workflow.name}}-modelpvc"
+								}
+							}
+						]
+					}
+				},
+				"tfReplicaType": "MASTER"
+			}
+		}
+	}
+}
+"""
+
+    tfjob = json.loads(tfjobjson)
+
+    train = dsl.ResourceOp(
+        name="train",
+        k8s_resource=tfjob,
+        success_condition='status.replicaStatuses.Worker.succeeded == 1',
+        attribute_outputs={"name": "{.metadata.name}"}
+    ).after(build).after(modelvolop)
```

Comment:
Since #1346, this can be written as `.after(build, modelvolop)`.
But if you decide to use `PipelineParam`s, as I explained above, you will just have to specify the `.after(build)` dependency.

[And if we close #1344 (sorry for the delay, I got held up), none of the two will be needed.]

Reply:
Thanks have made suggested change